src/OpenTelemetry/AssemblyInfo.cs

```diff
@@ -17,7 +17,6 @@
 using System.Runtime.CompilerServices;
 
 [assembly: InternalsVisibleTo("OpenTelemetry.Tests" + AssemblyInfo.PublicKey)]
-[assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.Dependencies.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.AspNetCore.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.StackExchangeRedis.Tests" + AssemblyInfo.PublicKey)]
```

Comment:
@CodeBlanch is this needed? If yes, this seems like a design flaw in the SDK - the SDK should provide sufficient interface for 3rd party integration to have proper unit test, without putting downstream assemblies in its trust/visibility boundary.

@cijothomas We probably should remove `AspNetCore.Tests` from this file, too.

Reply:
i think we can remove. At least, locally, when i removed it didn't show any error.

Reply:
I'll keep it for now as it is beyond the scope of this PR 😄

Reply:
@reyang No idea! I checked the history, looks like it has always been in there. As long as build and tests pass, removing sounds good to me.